pkg/R/RDD.R

```diff
@@ -1039,6 +1039,60 @@ setMethod("keyBy",
             lapply(rdd, apply.func)
           })
 
+#' Return a new RDD that has exactly numPartitions partitions.
+#' Can increase or decrease the level of parallelism in this RDD. Internally,
+#' this uses a shuffle to redistribute data.
+#' If you are decreasing the number of partitions in this RDD, consider using
+#' coalesce, which can avoid performing a shuffle.
+#'
+#' @param rdd The RDD.
+#' @param numPartitions Number of partitions to create.
+#' @rdname repartition
+#' @seealso coalesce
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, list(1, 2, 3, 4, 5, 6, 7), 4L)
+#' numPartitions(rdd)                   # 4
+#' numPartitions(repartition(rdd, 2L))  # 2
+#'}
+setGeneric("repartition", function(rdd, numPartitions) { standardGeneric("repartition") })
+
+#' @rdname repartition
+#' @aliases repartition,RDD
+setMethod("repartition",
+          signature(rdd = "RDD", numPartitions = "integer"),
+          function(rdd, numPartitions) {
+            jrdd <- callJMethod(getJRDD(rdd), "repartition", numPartitions)
+            RDD(jrdd)
+          })
+
+#' Return a new RDD that is reduced into numPartitions partitions.
+#'
+#' @param rdd The RDD.
+#' @param numPartitions Number of partitions to create.
+#' @rdname coalesce
+#' @seealso repartition
+#' @export
+#' @examples
+#'\dontrun{
+#' sc <- sparkR.init()
+#' rdd <- parallelize(sc, list(1, 2, 3, 4, 5), 3L)
+#' numPartitions(rdd)               # 3
+#' numPartitions(coalesce(rdd, 1L)) # 1
+#'}
+setGeneric("coalesce", function(rdd, numPartitions, ...) { standardGeneric("coalesce") })
+
+#' @rdname coalesce
+#' @aliases coalesce,RDD
+setMethod("coalesce",
+           signature(rdd = "RDD", numPartitions = "integer"),
```

Comment:
Does it make sense to accept `numPartitions` as "numeric" types, so that user can pass in something like `2` instead of `2L`? We can just cast them to integer inside.